rust/src/lib/ifaces/inter_ifaces.rs

```diff
@@ -176,6 +179,47 @@ impl Interfaces {
         Ok(())
     }
 
+    fn remove_unknown_type_port(&mut self) {
+        let mut pending_actions: Vec<(String, InterfaceType, String)> =
+            Vec::new();
+        for (iface_name, iface) in self.kernel_ifaces.iter() {
+            if !iface.is_controller() {
+                continue;
+            }
+            for port_name in find_unknown_type_port(iface, self) {
+                pending_actions.push((
+                    iface_name.to_string(),
+                    iface.iface_type(),
+                    port_name.to_string(),
+                ));
+            }
+        }
+        for iface in self.user_ifaces.values() {
+            if !iface.is_controller() {
+                continue;
+            }
+            for port_name in find_unknown_type_port(iface, self) {
+                pending_actions.push((
+                    iface.name().to_string(),
+                    iface.iface_type(),
+                    port_name.to_string(),
+                ));
+            }
```

Comment:
The for loop is repeated twice with almost identical body. It might be a good idea to extract it into separate function.

Reply:
Changed to use:

```rust
        for iface in
            self.kernel_ifaces.values()..chain(self.user_ifaces.values())
        {

```